com.unity.ml-agents/Tests/Editor/MLAgentsEditModeTest.cs

```diff
@@ -338,6 +339,7 @@ public void TestAgent()
                 {
                     agent2.LazyInitialize();
```

Comment:
```suggestion
                    // Since Agent2 is initialized after the Academy has stepped, its OnEpisodeBegin should be called now.
                    Assert.AreEqual(0, agent2.agentResetCalls);
                    agent2.LazyInitialize();
                    Assert.AreEqual(1, agent2.agentResetCalls);
```
(and get rid of tracking numberAgent2Reset, since it's harder to understand what we're checking with it)